source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,377 @@
+    printRollingWindow(cluster_stats_cache.success_name_, cluster_stats_cache.success_, out_str);
+    printRollingWindow(cluster_stats_cache.errors_name_, cluster_stats_cache.errors_, out_str);
+    printRollingWindow(cluster_stats_cache.timeouts_name_, cluster_stats_cache.timeouts_, out_str);
+    printRollingWindow(cluster_stats_cache.rejected_name_, cluster_stats_cache.rejected_, out_str);
+    printRollingWindow(cluster_stats_cache.total_name_, cluster_stats_cache.total_, out_str);
+  }
+  return out_str.str();
+}
+
+void HystrixSink::printRollingWindow(absl::string_view name, RollingWindow rolling_window,
+                                     std::stringstream& out_str) {
+  out_str << name << " | ";
+  for (auto specific_stat_vec_itr = rolling_window.begin();
+       specific_stat_vec_itr != rolling_window.end(); ++specific_stat_vec_itr) {
+    out_str << *specific_stat_vec_itr << " | ";
+  }
+  out_str << std::endl;
+}
+
+HystrixSink::HystrixSink(Server::Instance& server, const uint64_t num_of_buckets)
+    : // stats_(new HystrixStatCache(num_of_buckets)),
+      server_(server), current_index_(num_of_buckets), window_size_(num_of_buckets + 1) {
+  init();
+}
+
+HystrixSink::HystrixSink(Server::Instance& server)
+    : // stats_(new HystrixStatCache()),
+      server_(server), current_index_(DEFAULT_NUM_OF_BUCKETS),
+      window_size_(DEFAULT_NUM_OF_BUCKETS + 1) {
+  init();
+}
+
+void HystrixSink::init() {
+  Server::Admin& admin = server_.admin();
+  ENVOY_LOG(debug,
+            "adding hystrix_event_stream endpoint to enable connection to hystrix dashboard");
+  admin.addHandler("/hystrix_event_stream", "send hystrix event stream",
+                   MAKE_ADMIN_HANDLER(handlerHystrixEventStream), false, false);
+}
+
+Http::Code HystrixSink::handlerHystrixEventStream(absl::string_view,
+                                                  Http::HeaderMap& response_headers,
+                                                  Buffer::Instance&,
+                                                  Server::AdminStream& admin_stream) {
+
+  response_headers.insertContentType().value().setReference(
+      Http::Headers::get().ContentTypeValues.TextEventStream);
+  response_headers.insertCacheControl().value().setReference(
+      Http::Headers::get().CacheControlValues.NoCache);
+  response_headers.insertConnection().value().setReference(
+      Http::Headers::get().ConnectionValues.Close);
+  response_headers.insertAccessControlAllowHeaders().value().setReference(
+      Http::Headers::get().AccessControlAllowHeadersValue.AccessControlAllowHeadersHystrix);
+  response_headers.insertAccessControlAllowOrigin().value().setReference(
+      Http::Headers::get().AccessControlAllowOriginValue.All);
+  response_headers.insertNoChunks().value().setReference("0");
+
+  Http::StreamDecoderFilterCallbacks& stream_decoder_filter_callbacks =
+      const_cast<Http::StreamDecoderFilterCallbacks&>(admin_stream.getDecoderFilterCallbacks());
```

Comment:
Why do we need to const cast this?  If this interface returns a `const StreamDecoderFilterCallbacks&`, I think there's something wrong with it. `StreamDecoderFilterCallbacks` has no const methods, which means a const one would essentially be useless, right?  I think this should be updated to return a non-const `StreamDecoderFilterCallbacks&`.

Reply:
We wanted to return only const refs when we worked on the AdminStream. It forced me to use the const_cast for the hystrix implementation. We overlooked the fact there are no const methods on it anyway.